lib/browser/mqtt.ts

```diff
@@ -300,13 +295,16 @@ export class MqttClientConnection extends BufferedEventEmitter {
      *          true for resuming an existing session, or false if the session is new
      */
     async connect() {
+        setTimeout(() => { this.uncork() }, 0);
         return new Promise<boolean>((resolve, reject) => {
-            this.connection.once('connect', (connack: IConnackPacket) => {
-                resolve(connack.sessionPresent);
-            });
-            this.connection.once('error', (error: Error) => {
+            const on_connect_error = (error: Error) => {
                 reject(new CrtError(error));
+            };
+            this.connection.once('connect', (connack: mqtt.IConnackPacket) => {
+                this.connection.removeListener('error', on_connect_error);
+                resolve(connack.sessionPresent);
```

Comment:
Just curious why you added the `.removeListener()` call.
[Docs](https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/Promise.jsm/Promise) say resolve()/reject() do nothing if the promise was already fulfilled.
It is cleaner, but not strictly necessary.

Reply:
I added the removeListener() call because otherwise, the first time there's an error, this callback will trigger.

Reply:
ohhhhhhh yeah
👍